src/model/MediaDataSource.ts

```diff
@@ -20,6 +20,19 @@ export default class MediaDataSource extends MongoDataSource<MediaObject> {
     }
   }]
 
+  /**
+   * Find one media object by id.  Throw an exception if not found.
+   * @param _id
+   */
+  async getOneMediaObjectById (_id: mongoose.Types.ObjectId): Promise<MediaObject> {
+    const rs = await this.mediaObjectModel.find({ _id }).orFail(new Error('Media not found')).lean()
+    if (rs != null && rs.length === 1) {
+      return rs[0]
+    }
+    logger.error(`This shouldn't happend. Found multiple media objects for id: ${_id.toString()}`)
+    throw new Error('Media not found for id')
```

Comment:
I intentionally make the message slightly different than the case where there is 0 so we can debug the issue. And at the time, make it a bit vague in order to not scare users that we have corrupted data.  Since media._id is unique theoretically we should never get back more than 1 document.